plugins/flatpak/gs-flatpak-transaction.c

```diff
@@ -265,6 +302,26 @@ _transaction_operation_done (FlatpakTransaction *transaction,
 	}
 	switch (flatpak_transaction_operation_get_operation_type (operation)) {
 	case FLATPAK_TRANSACTION_OPERATION_INSTALL:
+		/* Handle special snowflake where "should-download" related refs for an installed ref
+		 * goes missing. In that case, libflatpak marks the main app ref as updatable
+		 * and then FlatpakTransaction resolves one of it's ops to install the related ref(s).
+		 *
+		 * We can depend on libflatpak till here. Since, libflatpak returns the main app
+		 * ref as updatable (instead of the related ref), we need to sync the main app's
+		 * state for UI/UX.
+		 *
+		 * Map the current op's ref (which is related ref) to it's main app ref (which is
+		 * currently shown in the UI) and set the state of the main GsApp object back
+		 * to INSTALLED here.
+		 *
+		 * This detection whether a related ref belongs to a main ref is quite sub-optimal as
```

Comment:
What's sub-optimal about it?

Reply:
I mean it's still querying related refs of the installed app, comparing with operation's current's ref and making sure the extension ref belong to the installed app's ref ... 

Seems like stitched-code.

Reply:
Yeah I suppose it could be cleaner, perhaps if the app object had the related refs directly associated with it.